couch/tests/conftest.py

```diff
@@ -47,24 +48,14 @@ def couch_cluster():
     env['COUCH_PORT'] = common.PORT
     couch_version = env["COUCH_VERSION"][0]
 
-    args = [
-        "docker-compose",
-        "-f", os.path.join(common.HERE, 'compose', 'compose_v{}.yaml'.format(couch_version))
-    ]
-
-    subprocess.check_call(args + ["down"])
-    subprocess.check_call(args + ["up", "-d"])
-
-    # wait for the cluster to be up before yielding
-    if not wait_for_couch():
-        subprocess.check_call(args + ["down"])
-        raise Exception("couchdb container boot timed out!")
-
-    generate_data(couch_version)
-
-    yield
-
-    subprocess.check_call(args + ["down"])
+    with docker_run(
+        compose_file=os.path.join(common.HERE, 'compose', 'compose_v{}.yaml'.format(couch_version)),
+        env_vars=env,
+        endpoints=[common.URL],
+    ):
+        generate_data(couch_version)
```

Comment:
It doesn't work

Reply:
I encountered this as well, it's because `endpoints` to check are appended to the list of conditions. So it will try to generate data before the container is even up, and fail.
Workaround is to use `CheckEndpoints` in `conditions` so that these are run before. Like this https://github.com/DataDog/integrations-core/pull/2969/files#diff-9ecd4440884b9b1ed2dccf7846fdd398R28

Reply:
Rhoo this is awesome!